### DISCUSSION TITLE
C1. Sunumu Ayarla (Kolay Versiyon)

### DISCUSSION THREAD

**Discussion by adasue** [sonsuzus/sonsuzus#498] [Soru Cevap]
n üye içeren bir takımın üyeleri, 1’den n’e numaralandırılmış, geniş bir toplantıda slayt sunumu yapmak için dizilmiştir. Slayt gösterisi m slayt içerir.
Uzunluğu n olan bir a dizisivardır. İlk olarak, üyeler bir sıra halinde ayakta durmaktadırlar a_1, a_2,….,a_n baştan sona doğru. Slayt gösterisi sırasıyla slayt 1Den m’ doğru sunulacaktır. Her bir bölüm sıranın başında bulunan üye tarafından sunulacaktır. Her bir slayt sunulduktan sonra, sıranın başında bulunan üyeyi sırada istediğin konuma getirebilirsiniz (diğer üyelerin sıralanış dizisini değiştirmeyecek şekilde). Örneğin, üyelerin sırasının [3,1,2,4] olduğunu düşün. 3. üye o anki slaytı sunduktan sonra, üyelerin sıralanışını [3,1,2,4], [1,3,2,4],[1,2,3,4] ya da [1,2,4,3] dizilerinden birine değiştirebilirsin. 
Aynı zamanda uzunluğu m olan bir b dizisi de vardır. Eğer slayt gösterisi üye b_i’nin bu koşullar altında 1Den m’e kadar olan bütün slayt i’leri sunmasını sağlayabiliyorsa iyi sayılır.
Fakat senin gıcık müdürün b dizisine q yenilik yapmak istiyor. i’inci yenilikte, slayt s_i ve üye t_i seçecek ve b_(s_i)=t_i haline getirecek. Bu yeniliklerin kalıcı olduğunu not alın, bu da demek oluyor ki b dizisinde yapılan değişiklikler ileriki yenilikler yapılırken de uygulanmaya devam edecek. 
Her bir q+1 durumları için olan b dizisinde, ilk durumda ve her bir q yenilikten sonra, slayt gösterisinin iyi olup olmadığına karar verin. 
Girdi
Her bir test birden fazla test senaryosu içerir. İlk kısım t sayıda test senaryosu içerir (1 küçük = t küçük eşittir 10 üzeri 4). Test senaryolarının tanımı şu şekildedir. 
Test senaryolarının ilk kısımları 3 tamsayı içerir n, m ve q (1 küçük = n,m küçük=2*10 üzeri 5;q=0)- üye sayısı, kısım sayısı ve yenilik sayısı.
Test senaryolarının ikinci kısmı n tamsayıları içerir a_1, a_2,....,a_n(1 küçük = a_i küçük = n)- üyelerin baştan sona doğru ilk sıralanışları. 1’den n’e kadar olan tamsayıların her birinin tam olarak bir kez a’da ortaya çıkacağı garantilenmiştir.
Her bir test senaryosunun 3. Kısmı m tamsayılarını içerir b_1, b_2,….,b_m (1 küçük = b_i küçük=n)- her bir kesiti sunması gereken üyeler.
Tüm test senaryolarındaki n ve m toplamlarının sırasıyla 2*10 üzeri 5’i aşmadığı garantilenmiştir.
Çıktı
Her bir test senaryosu için, b dizisinin q+1 durumuna karşılık gelen q+1 sıranın çıktısını al. Eğer slayt gösterisi iyise “YA”, değilse “TIDAK” yazınız.
Çıktı her bir durumda cevap üretebilir. Yazının büyüklüğü fark etmeksizin.

**Comment by adasue** [comment]
```
def cozum(): 
n, m, q = map(int, input().split()) 
a = list(map(int, input().split())) 
b = [] 
for i in range(m): 
x = int(input()) 
if len(b) == 0 or b[-1] != x: 
b.append(x) 
m = len(b) 
ziyaret_edilen = [0] * n 
j = 0 
for i in range(m): 
if j == n: 
print("YA") 
return 
if a[j] == b[i]: 
ziyaret_edilen[a[j] - 1] = 1 
continue 
else: 
if ziyaret_edilen[a[j] - 1] == 1: 
j += 1 
i -= 1 
else: 
print("TIDAK") 
return 
print("YA") 
t = int(input()) 
for _ in range(t): 
cozum()
```

**Reply by sonsuzus** [reply]
Kod çalışmıyor.

**Comment by adasue** [comment]
Hocam yarin isbo sınavım var,sınavdan sonra tekrar bakacağım.

1 Ara 2024 Pzr 12:02 tarihinde Sonsuz Us ***@***.***> şunu
yazdı:

**Reply by sonsuzus** [reply]
Başarılar sınavda.

**Comment by adasue** [comment]
Sağolun hocam

1 Ara 2024 Pzr 12:23 tarihinde Sonsuz Us ***@***.***> şunu
yazdı:
